lib/ruby_lsp/listeners/document_link.rb

```diff
@@ -124,7 +124,16 @@ def extract_document_link(node)
         match = comment.location.slice.match(%r{source://.*#\d+$})
         return unless match
 
-        uri = T.cast(URI(T.must(match[0])), URI::Source)
+        uri = T.cast(
+          begin
+            URI(T.must(match[0]))
+          rescue URI::InvalidURIError, URI::InvalidComponentError
```

Comment:
```suggestion
          rescue URI::Error
```
(covers these plus `URI::BadURIError`)